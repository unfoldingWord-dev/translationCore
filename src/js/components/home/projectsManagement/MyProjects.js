import React from 'react';
import PropTypes from 'prop-types';
// components
import ProjectCard from './ProjectCard';

let MyProjects = (props) => {
  const {myProjects} = props;
  const projects = myProjects.map( (projectDetails, index) =>
    <ProjectCard key={index} projectDetails={projectDetails} actions={props.actions} />
  );

  return (
    <div style={{ height: '100%' }}>
      Projects
      <div style={{ height: '100%', overflowY: 'auto', paddingRight: '10px' }}>
        {projects}
      </div>
    </div>
  );
}

MyProjects.propTypes = {
  myProjects: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired
};

export default MyProjects;
