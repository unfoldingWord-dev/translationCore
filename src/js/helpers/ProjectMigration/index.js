import migrateAppsToDotApps from './migrateAppsToDotApps';
import migrateToVersion1 from './migrateToVersion1';
import migrateToVersion2 from './migrateToVersion2';
import migrateToVersion3 from './migrateToVersion3';

/**
 * @Description:
 * Helpers for migrating projects but not specific to one migration
 * @param {String} projectSaveLocation - path to project
 * @param {string} link - Link to the projects git repo if provided i.e. https://git.door43.org/royalsix/fwe_tit_text_reg.git
 */

export default (projectSaveLocation, link) => {
  migrateAppsToDotApps(projectSaveLocation);
  migrateToVersion1(projectSaveLocation, link);
  migrateToVersion2(projectSaveLocation, link);
  migrateToVersion3(projectSaveLocation, link);
};
