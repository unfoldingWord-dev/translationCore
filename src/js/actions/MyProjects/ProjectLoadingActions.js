import consts from '../ActionTypes';
import path from 'path-extra';
// actions
import * as ProjectMigrationActions from '../Import/ProjectMigrationActions';
import * as ProjectValidationActions from '../Import/ProjectValidationActions';
import * as ToolsMetadataActions from '../ToolsMetadataActions';
import * as BodyUIActions from '../BodyUIActions';
import * as RecentProjectsActions from '../RecentProjectsActions';
import * as AlertModalActions from '../AlertModalActions';
import * as ProjectDetailsActions from '../ProjectDetailsActions';
import * as ProjectLoadingActions from '../MyProjects/ProjectLoadingActions';
import * as ProjectImportStepperActions from '../ProjectImportStepperActions';
//helpers
import * as manifestHelpers from '../../helpers/manifestHelpers';
// constants
const PROJECTS_PATH = path.join(path.homedir(), 'translationCore', 'projects');

/**
 * @description action that Migrates, Validates and Loads the Project
 * This may seem redundant to run migrations and validations each time
 * But the helpers called from each action test to only run when needed
 * @param {String} selectedProjectFilename
 */
export const migrateValidateLoadProject = (selectedProjectFilename) => {
  return(async (dispatch) => {
    try {
      dispatch(AlertModalActions.openAlertDialog('Loading your project data', true));
      setTimeout(async () => {
        let projectPath = path.join(PROJECTS_PATH, selectedProjectFilename);
        ProjectMigrationActions.migrate(projectPath);
        dispatch(AlertModalActions.closeAlertDialog());
        await dispatch(ProjectValidationActions.validate(projectPath));
        dispatch(displayTools());
      }, 200);
    } catch (error) {
      if (error.type !== 'div') console.warn(error);
      // clear last project must be called before any other action.
      // to avoid triggering autosaving.
      dispatch(clearLastProject());
      dispatch(AlertModalActions.openAlertDialog(error));
      dispatch(ProjectImportStepperActions.cancelProjectValidationStepper());
      dispatch({ type: "LOADED_ONLINE_FAILED" });
    }
  });
};

/**
 * @description - Opening the tools screen upon making sure the project is
 * not a titus or in the user is in developer
 */
export function displayTools() {
  return ((dispatch, getState) => {
    const { currentSettings } = getState().settingsReducer;
    const { manifest } = getState().projectDetailsReducer;
    if (manifestHelpers.checkIfValidBetaProject(manifest) || currentSettings.developerMode) {
      dispatch(ToolsMetadataActions.getToolsMetadatas());
      // Go to toolsCards page
      dispatch(BodyUIActions.goToStep(3));
    } else {
      dispatch(AlertModalActions.openAlertDialog('This version of translationCore only supports Titus projects.'));
      dispatch(RecentProjectsActions.getProjectsFromFolder());
    }
  });
}

/**
 * @description - Wrapper to clear everything in the store that could
 * prevent a new project from loading
 */
export function clearLastProject() {
  return ((dispatch) => {
    dispatch(BodyUIActions.toggleHomeView(true));
    dispatch(ProjectDetailsActions.resetProjectDetail());
    dispatch({ type: consts.CLEAR_PREVIOUS_GROUPS_DATA });
    dispatch({ type: consts.CLEAR_PREVIOUS_GROUPS_INDEX });
    dispatch({ type: consts.CLEAR_CONTEXT_ID });
    dispatch({ type: consts.CLEAR_CURRENT_TOOL_DATA });
    dispatch({ type: consts.CLEAR_RESOURCES_REDUCER });
    dispatch({
      type: consts.SET_CURRENT_TOOL_TITLE,
      currentToolTitle: ""
    });
    /** After clearing the local project the label also needs to be updated in the stepper */
    dispatch(BodyUIActions.resetStepLabels(1));
  });
}

/**
 * @description loads and set the projects details into the projectDetailsReducer.
 * @param {string} projectPath - path location in the filesystem for the project.
 * @param {object} manifest - project manifest.
 */
export function loadProjectDetails(projectPath, manifest) {
  return ((dispatch) => {
    dispatch(ProjectDetailsActions.setSaveLocation(projectPath));
    dispatch(ProjectDetailsActions.setProjectManifest(manifest));
  });
}
