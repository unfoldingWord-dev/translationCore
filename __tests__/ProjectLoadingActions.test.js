import configureMockStore from 'redux-mock-store';
import fs from 'fs-extra';
import thunk from 'redux-thunk';
import path from 'path-extra';
import ospath from 'ospath';
import * as ProjectLoadingActions from "../src/js/actions/MyProjects/ProjectLoadingActions";
import * as manifestUtils from "../src/js/helpers/ProjectMigration/manifestUtils";
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const PROJECTS_PATH = path.join(ospath.home(), 'translationCore', 'projects');

jest.mock('fs-extra');

jest.mock('../src/js/selectors', () => ({
  getActiveLocaleLanguage: () => {
    return {code: 'en'};
  },
  getTranslate: () => {
    return jest.fn((code) => {
      return code;
    });
  },
  getProjectSaveLocation: (state) => {
    return (state.projectDetailsReducer.projectSaveLocation);
  },
  getProjectManifest: (state) => {
    return (state.projectDetailsReducer.manifest);
  },
  getIsUserLoggedIn: (state) => {
    return (state.loginReducer.loggedInUser);
  },
  getActiveHomeScreenSteps: (state) => {
    const isLoggedIn = state.loginReducer.loggedInUser;
    const isProjectLoaded = state.projectDetailsReducer.projectSaveLocation;
    const availableSteps = [true, true, isLoggedIn, isProjectLoaded];
    return availableSteps;
  },
  getUsername: () => {
    return 'johndoe';
  }
}));

describe('ProjectLoadingActions.migrateValidateLoadProject', () => {
  let initialState = {};
  const projectName = 'en_tit';
  const sourcePath = path.join(__dirname, 'fixtures/project');
  const projectPath = path.join(PROJECTS_PATH, projectName);
  beforeEach(() => {
    // reset mock filesystem data
    fs.__resetMockFS();
    let copyFiles = [projectName];
    fs.__loadFilesIntoMockFs(copyFiles, sourcePath, PROJECTS_PATH);
    const manifest = manifestUtils.getProjectManifest(projectPath);

    initialState = {
      homeScreenReducer: {
        stepper: {
          stepIndex: 1,
          nextStepName: 'Project Information',
          previousStepName: 'Cancel',
          nextDisabled: false
        }
      },
      loginReducer: {
        loggedInUser: false,
        userdata: {},
        feedback: '',
        subject: 'Bug Report',
        placeholder: 'Leave us your feedback!'
      },
      projectDetailsReducer: {
        projectSaveLocation: projectPath,
        manifest,
        currentProjectToolsProgress: {},
        projectType: null
      },
      localImportReducer: {
        selectedProjectFilename: '',
        sourceProjectPath: '',
        oldSelectedProjectFileName: null
      },
      projectInformationCheckReducer: {alreadyImported:true},
      settingsReducer: { currentSettings: {}},
      projectValidationReducer: {
        projectValidationStepsArray: [ ]
      }
    };
  });

  it('should open to details screen if project name is not valid', async () => {
    // given
    const store = mockStore(initialState);
    const selectedProjectFilename = projectName;

    // when
    await store.dispatch(ProjectLoadingActions.migrateValidateLoadProject(selectedProjectFilename));

    //then
    expect(cleanupActions(store.getActions())).toMatchSnapshot();
  });

  it('should just open project if project name is valid', async () => {
    // given
    const selectedProjectFilename = 'en_ult_tit_book';
    const resourceId = "ult";
    // rename project to selected
    renameProject(selectedProjectFilename, projectPath, initialState, resourceId);

    const store = mockStore(initialState);

    // when
    await store.dispatch(ProjectLoadingActions.migrateValidateLoadProject(selectedProjectFilename));

    //then
    expect(cleanupActions(store.getActions())).toMatchSnapshot();
  });
});

describe('loadProject', () => {
  let initialState = {};
  const projectName = 'en_tit';
  const sourcePath = path.join(__dirname, 'fixtures/project');
  const projectPath = path.join(PROJECTS_PATH, projectName);
  beforeEach(() => {
    // reset mock filesystem data
    fs.__resetMockFS();
    let copyFiles = [projectName];
    fs.__loadFilesIntoMockFs(copyFiles, sourcePath, PROJECTS_PATH);
    const manifest = manifestUtils.getProjectManifest(projectPath);

    initialState = {
      homeScreenReducer: {
        stepper: {
          stepIndex: 1,
          nextStepName: 'Project Information',
          previousStepName: 'Cancel',
          nextDisabled: false
        }
      },
      loginReducer: {
        loggedInUser: false,
        userdata: {},
        feedback: '',
        subject: 'Bug Report',
        placeholder: 'Leave us your feedback!'
      },
      projectDetailsReducer: {
        projectSaveLocation: projectPath,
        manifest,
        currentProjectToolsProgress: {},
        projectType: null
      },
      localImportReducer: {
        selectedProjectFilename: '',
        sourceProjectPath: '',
        oldSelectedProjectFileName: null
      },
      projectInformationCheckReducer: {alreadyImported:true},
      settingsReducer: { currentSettings: {}},
      projectValidationReducer: {
        projectValidationStepsArray: [ ]
      }
    };
  });

  it('displays the details screen of an invalid project', async () => {
    // given
    const store = mockStore(initialState);

    // when
    await store.dispatch(ProjectLoadingActions.openProject(projectName));

    //then
    expect(cleanupActions(store.getActions())).toMatchSnapshot();
  });

  it('opens a valid project', async () => {
    // given
    const selectedProjectFilename = 'en_ult_tit_book';
    const resourceId = "ult";
    // rename project to selected
    renameProject(selectedProjectFilename, projectPath, initialState, resourceId);

    const store = mockStore(initialState);

    // when
    await store.dispatch(ProjectLoadingActions.openProject(selectedProjectFilename));

    //then
    expect(cleanupActions(store.getActions())).toMatchSnapshot();
  });
});

//
// helpers
//

function cleanupActions(actions) {
  if (actions && actions.length) {
    for (const action of actions) {
      if (action.projects && Array.isArray(action.projects)) {
        for (const project of action.projects) {
          if (project.accessTimeAgo) {
            project.accessTimeAgo = 'now';
          }
        }
      }
      if ('pathLocation' in action) {
        action.pathLocation = path.basename(action.pathLocation);
      }
    }
  }
  return actions;
}

function renameProject(selectedProjectFilename, projectPath, initialState, resourceId) {
  const currentProjectPath = path.join(PROJECTS_PATH, selectedProjectFilename);
  fs.moveSync(projectPath, currentProjectPath);
  initialState.projectDetailsReducer.projectSaveLocation = currentProjectPath;
  initialState.projectDetailsReducer.manifest.resource.id = resourceId;
  fs.outputJsonSync(manifestUtils.getManifestPath(currentProjectPath), initialState.projectDetailsReducer.manifest);
}

