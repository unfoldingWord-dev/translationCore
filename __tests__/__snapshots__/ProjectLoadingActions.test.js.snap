// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLoadingActions.migrateValidateLoadProject should just open project if project name is valid 1`] = `
Array [
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "alertMessage": "projects.loading_project_alert",
    "loading": true,
    "type": "OPEN_ALERT_DIALOG",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "boolean": true,
    "type": "TOGGLE_HOME_VIEW",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_DATA",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_INDEX",
  },
  Object {
    "type": "CLEAR_CONTEXT_ID",
  },
  Object {
    "type": "CLEAR_CURRENT_TOOL_DATA",
  },
  Object {
    "type": "CLEAR_RESOURCES_REDUCER",
  },
  Object {
    "type": "CLEAR_PREVIOUS_FILTERS",
  },
  Object {
    "currentToolTitle": "",
    "type": "SET_CURRENT_TOOL_TITLE",
  },
  Object {
    "alertMessage": "project_validation.old_project_unsupported",
    "loading": undefined,
    "type": "OPEN_ALERT_DIALOG",
  },
  Object {
    "showProjectValidationStepper": false,
    "type": "TOGGLE_PROJECT_VALIDATION_STEPPER",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "boolean": true,
    "type": "TOGGLE_HOME_VIEW",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_DATA",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_INDEX",
  },
  Object {
    "type": "CLEAR_CONTEXT_ID",
  },
  Object {
    "type": "CLEAR_CURRENT_TOOL_DATA",
  },
  Object {
    "type": "CLEAR_RESOURCES_REDUCER",
  },
  Object {
    "type": "CLEAR_PREVIOUS_FILTERS",
  },
  Object {
    "currentToolTitle": "",
    "type": "SET_CURRENT_TOOL_TITLE",
  },
  Object {
    "type": "CLEAR_COPYRIGHT_CHECK_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "type": "CLEAR_MERGE_CONFLICTS_REDUCER",
  },
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "projects": Array [
      Object {
        "accessTimeAgo": "now",
        "bookAbbr": "tit",
        "bookName": "Titus",
        "isSelected": true,
        "projectName": "en_ult_tit_book",
        "projectSaveLocation": "<HOME_DIR>/translationCore/projects/en_ult_tit_book",
        "target_language": Object {
          "direction": "ltr",
          "id": "en",
          "name": "English",
        },
      },
    ],
    "type": "GET_MY_PROJECTS",
  },
  Object {
    "type": "LOADED_ONLINE_FAILED",
  },
]
`;

exports[`ProjectLoadingActions.migrateValidateLoadProject should open to details screen if project name is not valid 1`] = `
Array [
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "alertMessage": "projects.loading_project_alert",
    "loading": true,
    "type": "OPEN_ALERT_DIALOG",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "boolean": true,
    "type": "TOGGLE_HOME_VIEW",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_DATA",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_INDEX",
  },
  Object {
    "type": "CLEAR_CONTEXT_ID",
  },
  Object {
    "type": "CLEAR_CURRENT_TOOL_DATA",
  },
  Object {
    "type": "CLEAR_RESOURCES_REDUCER",
  },
  Object {
    "type": "CLEAR_PREVIOUS_FILTERS",
  },
  Object {
    "currentToolTitle": "",
    "type": "SET_CURRENT_TOOL_TITLE",
  },
  Object {
    "alertMessage": "project_validation.old_project_unsupported",
    "loading": undefined,
    "type": "OPEN_ALERT_DIALOG",
  },
  Object {
    "showProjectValidationStepper": false,
    "type": "TOGGLE_PROJECT_VALIDATION_STEPPER",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "boolean": true,
    "type": "TOGGLE_HOME_VIEW",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_DATA",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_INDEX",
  },
  Object {
    "type": "CLEAR_CONTEXT_ID",
  },
  Object {
    "type": "CLEAR_CURRENT_TOOL_DATA",
  },
  Object {
    "type": "CLEAR_RESOURCES_REDUCER",
  },
  Object {
    "type": "CLEAR_PREVIOUS_FILTERS",
  },
  Object {
    "currentToolTitle": "",
    "type": "SET_CURRENT_TOOL_TITLE",
  },
  Object {
    "type": "CLEAR_COPYRIGHT_CHECK_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "type": "CLEAR_MERGE_CONFLICTS_REDUCER",
  },
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "projects": Array [
      Object {
        "accessTimeAgo": "now",
        "bookAbbr": "tit",
        "bookName": "Titus",
        "isSelected": true,
        "projectName": "en_tit",
        "projectSaveLocation": "<HOME_DIR>/translationCore/projects/en_tit",
        "target_language": Object {
          "direction": "ltr",
          "id": "en",
          "name": "English",
        },
      },
    ],
    "type": "GET_MY_PROJECTS",
  },
  Object {
    "type": "LOADED_ONLINE_FAILED",
  },
]
`;

exports[`loadProject displays the details screen of an invalid project 1`] = `
Array [
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "alertMessage": "projects.loading_project_alert",
    "loading": true,
    "type": "OPEN_ALERT_DIALOG",
  },
  Object {
    "type": "CLOSE_ALERT_DIALOG",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "boolean": true,
    "type": "TOGGLE_HOME_VIEW",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_DATA",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_INDEX",
  },
  Object {
    "type": "CLEAR_CONTEXT_ID",
  },
  Object {
    "type": "CLEAR_CURRENT_TOOL_DATA",
  },
  Object {
    "type": "CLEAR_RESOURCES_REDUCER",
  },
  Object {
    "type": "CLEAR_PREVIOUS_FILTERS",
  },
  Object {
    "currentToolTitle": "",
    "type": "SET_CURRENT_TOOL_TITLE",
  },
  Object {
    "pathLocation": "en_tit",
    "type": "SET_SAVE_PATH_LOCATION",
  },
  Object {
    "manifest": Object {
      "checkers": Array [],
      "generator": Object {
        "build": "",
        "name": "tc-desktop",
      },
      "license": "CC BY-SA 4.0",
      "project": Object {
        "id": "tit",
        "name": "Titus",
      },
      "repo": "",
      "resource": Object {
        "id": "",
        "name": "",
      },
      "source_translations": Array [
        Object {
          "checking_level": "",
          "date_modified": "2018-01-31T19:19:27.914Z",
          "language_id": "en",
          "resource_id": "ult",
          "version": "",
        },
      ],
      "target_language": Object {
        "book": Object {
          "name": "Titus",
        },
        "direction": "ltr",
        "id": "en",
        "name": "English",
      },
      "tcInitialized": true,
      "tc_version": 6,
      "time_created": "2018-01-31T19:19:27.914Z",
      "tools": Array [],
      "translators": Array [],
      "ts_project": Object {
        "id": "tit",
        "name": "Titus",
      },
      "type": Object {
        "id": "text",
        "name": "Text",
      },
    },
    "type": "STORE_MANIFEST",
  },
  Object {
    "languageName": "English",
    "type": "SET_LANGUAGE_NAME_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageId": "en",
    "type": "SET_LANGUAGE_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageDirection": "ltr",
    "type": "SET_LANGUAGE_DIRECTION_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "bookId": "tit",
    "type": "SET_BOOK_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "resourceId": "",
    "type": "SET_RESOURCE_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "nickname": "",
    "type": "SET_NICKNAME_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "contributors": Array [],
    "type": "SET_CONTRIBUTORS_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "checkers": Array [],
    "type": "SET_CHECKERS_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "buttonName": "project_information",
    "index": 1,
    "namespace": "projectInformationCheck",
    "type": "ADD_PROJECT_VALIDATION_STEP",
  },
  Object {
    "pathLocation": "en_tit",
    "type": "SET_SAVE_PATH_LOCATION",
  },
  Object {
    "manifest": Object {
      "checkers": Array [],
      "generator": Object {
        "build": "",
        "name": "tc-desktop",
      },
      "license": "CC BY-SA 4.0",
      "project": Object {
        "id": "tit",
        "name": "Titus",
      },
      "repo": "",
      "resource": Object {
        "id": "",
        "name": "",
      },
      "source_translations": Array [
        Object {
          "checking_level": "",
          "date_modified": "2018-01-31T19:19:27.914Z",
          "language_id": "en",
          "resource_id": "ult",
          "version": "",
        },
      ],
      "target_language": Object {
        "book": Object {
          "name": "Titus",
        },
        "direction": "ltr",
        "id": "en",
        "name": "English",
      },
      "tcInitialized": true,
      "tc_version": 6,
      "time_created": "2018-01-31T19:19:27.914Z",
      "tools": Array [],
      "translators": Array [],
      "ts_project": Object {
        "id": "tit",
        "name": "Titus",
      },
      "type": Object {
        "id": "text",
        "name": "Text",
      },
    },
    "type": "STORE_MANIFEST",
  },
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageName": "English",
    "type": "SET_LANGUAGE_NAME_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageId": "en",
    "type": "SET_LANGUAGE_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageDirection": "ltr",
    "type": "SET_LANGUAGE_DIRECTION_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "bookId": "tit",
    "type": "SET_BOOK_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "resourceId": "",
    "type": "SET_RESOURCE_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "nickname": "",
    "type": "SET_NICKNAME_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "contributors": Array [],
    "type": "SET_CONTRIBUTORS_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "checkers": Array [],
    "type": "SET_CHECKERS_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "buttonName": "project_information",
    "index": 1,
    "namespace": "projectInformationCheck",
    "type": "ADD_PROJECT_VALIDATION_STEP",
  },
  Object {
    "type": "ONLY_SHOW_PROJECT_INFORMATION_SCREEN",
    "value": true,
  },
  Object {
    "showProjectValidationStepper": false,
    "type": "TOGGLE_PROJECT_VALIDATION_STEPPER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "type": "CLOSE_ALERT_DIALOG",
  },
  Object {
    "stepIndex": 3,
    "type": "GO_TO_STEP",
  },
]
`;

exports[`loadProject opens a valid project 1`] = `
Array [
  Object {
    "type": "RESET_PROJECT_VALIDATION_REDUCER",
  },
  Object {
    "type": "CLEAR_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "alertMessage": "projects.loading_project_alert",
    "loading": true,
    "type": "OPEN_ALERT_DIALOG",
  },
  Object {
    "type": "CLOSE_ALERT_DIALOG",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "boolean": true,
    "type": "TOGGLE_HOME_VIEW",
  },
  Object {
    "type": "RESET_PROJECT_DETAIL",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_DATA",
  },
  Object {
    "type": "CLEAR_PREVIOUS_GROUPS_INDEX",
  },
  Object {
    "type": "CLEAR_CONTEXT_ID",
  },
  Object {
    "type": "CLEAR_CURRENT_TOOL_DATA",
  },
  Object {
    "type": "CLEAR_RESOURCES_REDUCER",
  },
  Object {
    "type": "CLEAR_PREVIOUS_FILTERS",
  },
  Object {
    "currentToolTitle": "",
    "type": "SET_CURRENT_TOOL_TITLE",
  },
  Object {
    "pathLocation": "en_ult_tit_book",
    "type": "SET_SAVE_PATH_LOCATION",
  },
  Object {
    "manifest": Object {
      "checkers": Array [],
      "generator": Object {
        "build": "",
        "name": "tc-desktop",
      },
      "license": "CC BY-SA 4.0",
      "project": Object {
        "id": "tit",
        "name": "Titus",
      },
      "repo": "",
      "resource": Object {
        "id": "ult",
        "name": "",
      },
      "source_translations": Array [
        Object {
          "checking_level": "",
          "date_modified": "2018-01-31T19:19:27.914Z",
          "language_id": "en",
          "resource_id": "ult",
          "version": "",
        },
      ],
      "target_language": Object {
        "book": Object {
          "name": "Titus",
        },
        "direction": "ltr",
        "id": "en",
        "name": "English",
      },
      "tcInitialized": true,
      "tc_version": 6,
      "time_created": "2018-01-31T19:19:27.914Z",
      "tools": Array [],
      "translators": Array [],
      "ts_project": Object {
        "id": "tit",
        "name": "Titus",
      },
      "type": Object {
        "id": "text",
        "name": "Text",
      },
    },
    "type": "STORE_MANIFEST",
  },
  Object {
    "languageName": "English",
    "type": "SET_LANGUAGE_NAME_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageId": "en",
    "type": "SET_LANGUAGE_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "languageDirection": "ltr",
    "type": "SET_LANGUAGE_DIRECTION_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "bookId": "tit",
    "type": "SET_BOOK_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "resourceId": "ult",
    "type": "SET_RESOURCE_ID_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "nickname": "",
    "type": "SET_NICKNAME_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "contributors": Array [],
    "type": "SET_CONTRIBUTORS_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "checkers": Array [],
    "type": "SET_CHECKERS_IN_PROJECT_INFORMATION_REDUCER",
  },
  Object {
    "nextDisabled": true,
    "type": "UPDATE_PROJECT_VALIDATION_NEXT_BUTTON_STATUS",
  },
  Object {
    "type": "CLOSE_ALERT_DIALOG",
  },
  Object {
    "stepIndex": 3,
    "type": "GO_TO_STEP",
  },
]
`;
